src/components/Reports/ReportsBox/ReportsBox.js

```diff
@@ -0,0 +1,74 @@
+import React from 'react';
+import { InjectAppServices } from '../../../services/pure-di';
+import { FormattedMessage } from 'react-intl';
+import './ReportsBox.css';
+
+class ReportsBox extends React.Component {
+  constructor({ dependencies: { datahubClient } }) {
+    super();
+
+    /** @type { import('../../services/datahub-client').DatahubClient } */
+    this.datahubClient = datahubClient;
+
+    this.state = {
+      visits: 0,
+      dateFrom: null,
+      dateTo: new Date().toLocaleDateString(),
+    };
+
+    this.getVisitsByPeriod = this.getVisitsByPeriod.bind(this);
+  }
+
+  async getVisitsByPeriod(domainName, periodSelectedDays) {
+    let dateFrom = new Date();
+    dateFrom.setDate(dateFrom.getDate() - parseInt(periodSelectedDays));
+    const visits = await this.datahubClient.getVisitsByPeriod(domainName, dateFrom);
+    if (visits) {
+      this.setState({
+        visits: visits,
+        dateFrom: new Date(dateFrom).toLocaleDateString(),
+      });
+    }
+  }
+
+  componentDidMount() {
+    this.getVisitsByPeriod(this.props.domainName, this.props.periodSelectedDays);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    this.getVisitsByPeriod(nextProps.domainName, nextProps.periodSelectedDays);
+  }
+
+  render() {
+    const {
+      state: { visits, dateFrom, dateTo },
+    } = this;
+    const isDataLoaded = !!visits;
```

Comment:
What do you think about adding a new value to the state about loading status?

Because, maybe `!visits` means something different than _loading status_, maybe an error.

Reply:
fixed